src/gtest/test_sidechain_to_mempool.cpp

```diff
@@ -131,7 +134,7 @@ class SidechainsInMempoolTestSuite: public ::testing::Test {
 
     void InitCoinGeneration();
     std::pair<uint256, CCoinsCacheEntry> GenerateCoinsAmount(const CAmount & amountToGenerate);
-    bool StoreCoins(const std::pair<uint256, CCoinsCacheEntry>& entryToStore);
```

Comment:
After the last changes, we noticed that the `StoreCoins()` function was taking the reference and destroying it.

Such a behavior made the unit test fail since the `entryToStore` argument was needed after the call itself.